youtube_dl/extractor/common.py

```diff
@@ -404,21 +404,27 @@ def __init__(self, downloader=None):
         self.set_downloader(downloader)
 
     @classmethod
-    def suitable(cls, url):
-        """Receives a URL and returns True if suitable for this IE."""
+    def _match_valid_url(cls, url):
+        """Receives a URL and returns match against the IE's _VALID_URL."""
 
         # This does not use has/getattr intentionally - we want to know whether
         # we have cached the regexp for *this* class, whereas getattr would also
         # match the superclass
         if '_VALID_URL_RE' not in cls.__dict__:
             cls._VALID_URL_RE = re.compile(cls._VALID_URL)
-        return cls._VALID_URL_RE.match(url) is not None
+        return cls._VALID_URL_RE.match(url)
+
+    @classmethod
+    def suitable(cls, url):
+        """Receives a URL and returns True if suitable for this IE."""
+
+        return cls._match_valid_url(url) is not None
```

Comment:
idk what the maintainers think about this, but I personally think this change is out of the scope of this PR. If this function is desired, it can be added separately. For now, you could just replace `self._match_valid_url(url)` with `re.match(self._VALID_URL, url)` as many other extractors already do.

Reply:
And I've done it myself. I thought this was a useful back-port although I agree that it might be controversial in the scope of the PR. I don't see any obvious downside to the change, although it would be a different case if yt-dlp was intending to revert this code (or wished it could).

One could just introduce a `CBCGemBaseIE` with the `_match_valid_url()` class method, as you've defined, so minimising code divergence from the yt-dlp PR, but then other extractors don't get the benefit.

Of course this is moot as long as no-one is accepting PRs (apparently there's almost a month of PRs to review at an average 15 minutes per PR in 8-hour days of nothing else but the odd coffee break), but I'll be merging it into my release.